### DISCUSSION TITLE
Ability to choose type of performance

### DISCUSSION THREAD

**Discussion by Biermannek** [ghostfolio/ghostfolio#2960] [Ideas]
Hello,

do you plan to return back "simple" performance for holdings? To be honest, I had to stick with 2.32.0 because that's the last version before Ghostfolio switched to time-weighted performance which in fact is total non-sense for continuous or DCA investments. It really doesn't make any sense for me to calculate performance from my first investment to particular asset. It doesn't give any value when you see that you have loss of XX% when you can see that value of your position is higher than it's price.

From all possible performance calculations I can think of (simple, time-weighter, money-weighted, XIRR), the current one is worst and I really don't understand why it was changed this way.

**Comment by dtslvr** [comment]
Hi @Biermannek

Thanks for sharing your thoughts on Ghostfolio's performance calculations! A great starting point would be to lay the groundwork by implementing a strategy pattern for multiple algorithms. In a second step, feel free to add the calculation method that aligns with your preference. Looking forward to your enhancements!

**Reply by Biermannek** [reply]
Hi,
thank you for your response. For me it would be totally enough if it would be possible to choose between performance calculation used up to 2.32.0 and that time-weighted which was implemented in 2.33.0, so I think both calculations are already implemented in some branch code :)

If I understand it correcty, then these are the versions:
1) Up to 2.32.0 - Performance is calculated simply as ratio between price of assets and current value of assets. If I bought it for 2000 and current price is 3000, then performance is +50%. This is totally fine
2) From 2.33.0 - Performance is calculated from price of the first buy (or first day of chosen period?) and current price. So if I bought 1 unit of stock a year ago for 100 and current price is 150, then performance is +50% regardless of other purchases of the same stock

So if I take an example:
1) 1.1.2023 - bought 1 unit for $100
2) 1.6.2023 - bought 100 units for $50
3) 1.1.2024 - price is $90 and I own 101 units

So in version 2.32.0 I will see that I own assets worth $9090, I paid $5100 for them and performance is +78%
In version 2.33 and newer I will see performace -10% if I'm right, because price has dropped by 10% since 1.1.2023 and additional purchases doesn't influence time-weighted performance as far as I know  (correct me if I'm wrong, as I wrote, I have downgraded so can't retest this calculation)

**Reply by karter16** [reply]
Yeah time-weighted returns are not a particularly useful measure for an individual investor who is in control of inflows and outflows. This article explains why well.

https://www.sharesight.com/blog/time-weighted-vs-money-weighted-rates-of-return/

as part of this future work it would be great to have a MWR option available as well.

Interestingly I tested this out in 2.46 and what it implements is not actually TWR, it’s something else again….

**Comment by ups7** [comment]
Is it possible to provide a switch between time-weighted or the previous calculation via an environment variable or a toggle in the admin panel/user settings? I understand that time-weighted calculation can be of benefit to some but often I just want to see a direct P/L calculation which Ghostfolio used to support until recently. Assuming the code for the previous calculation logic hasn't been lost forever, will it be too much to ask and too much work to have both options but allow the user to choose whichever they prefer?

**Reply by x1c0** [reply]
I think this is the best approach, I have the same requirement, for me personally I would like just to see the direct performance like I see in Interactive Brokers (portfolio table view)

I'll try to make a PR for this :-)

**Comment by karter16** [comment]
Please, please, please can we have something, anything, other than time-weighted performance. For the purposes of an individual investor tracking the performance of their portfolio time-weighted returns are entirely useless.

**Reply by fib1985** [reply]
I agree

**Comment by soloam** [comment]
I think that both performances have their use cases! And I would be happy to have a extra column in the Holdings with the actual profit percentage!

Percentage of Profit = ( (Current Value - Average Value) / Average Value ) * 100

**Comment by underline9396** [comment]
I ended up forking the repo and changing the performance calculation to `netPerformance / totalInvestmentValue`.  Unfortunately I cannot use this project with `timeWeightedInvestmentValue`

**Reply by dtslvr** [reply]
Hi @underline9396

As the [foundation](https://github.com/ghostfolio/ghostfolio/tree/main/apps/api/src/app/portfolio/calculator) for different performance calculations is now in place, would you be interested in contributing your implementation back to the project as a new algorithm option? This would allow other users to benefit from your work. Let us know what you think!

**Reply by underline9396** [reply]
Hey, thank you, I need a lot of time to catch up, I will try to look into it.

**Comment by ThowZzy** [comment]
I can't wait to see this implemented ! I also find this current performance calculation confusing for me. I only care about the raw performance of my portfolio at a given time.

**Comment by xiaojinlii** [comment]
I also hope to launch this feature. For individual investors, MWR or XIRR would be more useful.

**Comment by rafal111-true** [comment]
It would be great to have that option. Is that case dead?

**Comment by Mirage1337** [comment]
Also very interested on if there is still anything ongoing.
